hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordReader.java

```diff
@@ -230,6 +242,34 @@ public void processNextRecord(T record, Map<String, Object> metadata) throws IOE
     }
   }
 
+  @Override
+  public void processNextRecordWithPosition(T record,
+                                            Map<String, Object> metadata,
+                                            int sequenceNo,
+                                            List<Integer> positions) throws IOException {
+    String key = (String) metadata.get(HoodieReaderContext.INTERNAL_META_RECORD_KEY);
+    int position = positions.get(sequenceNo);
+    // No position information is found for this record.
+    if (position < 0) {
+      throw new IOException("No position information is found for the record " + key);
+    }
+
+    Pair<Option<T>, Map<String, Object>> existingRecordMetadataPair = positionToRecords.get(position);
+    if (existingRecordMetadataPair != null) {
+      HoodieRecord<T> combinedRecord = (HoodieRecord<T>) recordMerger.merge(
+          readerContext.constructHoodieRecord(Option.of(record), metadata, readerSchema),
+          readerSchema,
+          readerContext.constructHoodieRecord(existingRecordMetadataPair.getLeft(), existingRecordMetadataPair.getRight(), readerSchema),
+          readerSchema,
+          this.getPayloadProps()).get().getLeft();
```

Comment:
So payload will determine whether event time order is needed or not. Please check my comment regarding `useEventOrdering` boolean argument in `prepareRecordPositions` method.

Reply:
I saw the Ethan's PR, which is based on precombine field name. Which one should I use?

Reply:
@danny0405 Please chime in.

Reply:
We don't need to worry about this now since we assume this should be taken care of by the merger implementation.